From: "Kalderon, Michal" <Michal.Kalderon@cavium.com>
Date: Wed, 20 Sep 2017 08:39:46 +0300
Subject: RDMA/qedr: Fix rdma_type initialization
Patch-mainline: v4.15-rc1
Git-commit: e538e0aca33f81fbfaa10a6954b40ff4360686dd
References: bsc#1050545 FATE#322893

Initialize the rdma_type (iWARP or RoCE) which is set according to
device configuration in qed.

Fixes: e6a38c54faf ("RDMA/qedr: Add support for registering an iWARP device")

Signed-off-by: Michal Kalderon <Michal.Kalderon@cavium.com>
Signed-off-by: Ariel Elior <Ariel.Elior@cavium.com>
Signed-off-by: Doug Ledford <dledford@redhat.com>
Acked-by: Thomas Bogendoerfer <tbogendoerfer@suse.de>
---
 drivers/infiniband/hw/qedr/main.c |    1 +
 1 file changed, 1 insertion(+)

--- a/drivers/infiniband/hw/qedr/main.c
+++ b/drivers/infiniband/hw/qedr/main.c
@@ -870,6 +870,7 @@ static struct qedr_dev *qedr_add(struct
 		goto init_err;
 
 	dev->user_dpm_enabled = dev_info.user_dpm_enabled;
+	dev->rdma_type = dev_info.rdma_type;
 	dev->num_hwfns = dev_info.common.num_hwfns;
 	dev->rdma_ctx = dev->ops->rdma_get_rdma_ctx(cdev);
 
